crates/btc-relay/src/lib.rs

```diff
@@ -305,24 +215,304 @@ decl_module! {
+        StorageMap<_, Blake2_128Concat, H256Le, RichBlockHeader<T::AccountId, T::BlockNumber>, ValueQuery>;
+
+    /// Priority queue of BlockChain elements, ordered by the maximum height (descending).
+    /// The first index into this mapping (0) is considered to be the longest chain. The value
+    /// of the entry is the index into `ChainsIndex` to retrieve the `BlockChain`.
+    #[pallet::storage]
+    pub(super) type Chains<T: Config> = StorageMap<_, Blake2_128Concat, u32, u32>;
+
+    /// Auxiliary mapping of chains ids to `BlockChain` entries. The first index into this
+    /// mapping (0) is considered to be the Bitcoin main chain.
+    #[pallet::storage]
+    pub(super) type ChainsIndex<T: Config> = StorageMap<_, Blake2_128Concat, u32, BlockChain>;
+
+    /// Stores a mapping from (chain_index, block_height) to block hash
+    #[pallet::storage]
+    pub(super) type ChainsHashes<T: Config> =
+        StorageDoubleMap<_, Blake2_128Concat, u32, Blake2_128Concat, u32, H256Le, ValueQuery>;
+
+    /// Store the current blockchain tip
+    #[pallet::storage]
+    pub(super) type BestBlock<T: Config> = StorageValue<_, H256Le, ValueQuery>;
+
+    /// Store the height of the best block
+    #[pallet::storage]
+    pub(super) type BestBlockHeight<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+    /// BTC height when the relay was initialized
+    #[pallet::storage]
+    pub(super) type StartBlockHeight<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+    /// Increment-only counter used to track new BlockChain entries
+    #[pallet::storage]
+    pub(super) type ChainCounter<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+    /// Global security parameter k for stable Bitcoin transactions
+    #[pallet::storage]
+    #[pallet::getter(fn bitcoin_confirmations)]
+    pub(super) type StableBitcoinConfirmations<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+    /// Global security parameter k for stable Parachain transactions
+    #[pallet::storage]
+    #[pallet::getter(fn parachain_confirmations)]
+    pub(super) type StableParachainConfirmations<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+
+    /// Whether the module should perform difficulty checks.
+    #[pallet::storage]
+    #[pallet::getter(fn disable_difficulty_check)]
+    pub(super) type DisableDifficultyCheck<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+    /// Whether the module should perform inclusion checks.
+    #[pallet::storage]
+    #[pallet::getter(fn disable_inclusion_check)]
+    pub(super) type DisableInclusionCheck<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+    /// Whether the module should perform OP_RETURN checks.
+    #[pallet::storage]
+    #[pallet::getter(fn disable_op_return_check)]
+    pub(super) type DisableOpReturnCheck<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        #[doc = " Global security parameter k for stable Bitcoin transactions"]
```

Comment:
Can this the outer derive macro (`#[pallet::genesis_config]`) extract these from doc comments or does it have to be specified like this?